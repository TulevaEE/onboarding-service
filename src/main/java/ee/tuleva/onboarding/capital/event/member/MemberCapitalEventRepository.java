package ee.tuleva.onboarding.capital.event.member;

import java.math.BigDecimal;
import java.util.List;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;

public interface MemberCapitalEventRepository extends CrudRepository<MemberCapitalEvent, Long> {

  @Query("select sum(e.ownershipUnitAmount) from MemberCapitalEvent e")
  BigDecimal getTotalOwnershipUnitAmount();

  List<MemberCapitalEvent> findAllByMemberId(Long memberId);

  @Query(
      "select sum(e.fiatValue) from MemberCapitalEvent e where e.member.id = :memberId and e.type = :type")
  BigDecimal getTotalFiatValueByMemberIdAndType(Long memberId, MemberCapitalEventType type);

  @Query(
      "select sum(e.ownershipUnitAmount) from MemberCapitalEvent e where e.member.id = :memberId and e.type = :type")
  BigDecimal getTotalOwnershipUnitsByMemberIdAndType(Long memberId, MemberCapitalEventType type);

  List<MemberCapitalEvent> findAll();
}
