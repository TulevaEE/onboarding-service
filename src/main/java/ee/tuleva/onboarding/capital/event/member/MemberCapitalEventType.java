package ee.tuleva.onboarding.capital.event.member;

public enum MemberCapitalEventType {
  CAPITAL_PAYMENT,
  UNVESTED_WORK_COMPENSATION,
  WORK_COMPENSATION,
  MEMBERSHIP_BONUS,
}
