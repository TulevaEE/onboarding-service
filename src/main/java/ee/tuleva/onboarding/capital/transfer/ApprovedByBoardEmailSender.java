package ee.tuleva.onboarding.capital.transfer;

import static ee.tuleva.onboarding.capital.transfer.CapitalTransferContractState.APPROVED;
import static ee.tuleva.onboarding.capital.transfer.CapitalTransferContractState.APPROVED_AND_NOTIFIED;
import static ee.tuleva.onboarding.mandate.email.persistence.EmailType.CAPITAL_TRANSFER_APPROVED_BY_BOARD;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.annotation.Profile;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
@Profile("!dev")
@Slf4j
public class ApprovedByBoardEmailSender {

  private final CapitalTransferContractRepository capitalTransferContractRepository;
  private final CapitalTransferContractService capitalTransferContractService;

  // every 5 minutes
  @Scheduled(cron = "0 */5 * * * *", zone = "Europe/Tallinn")
  public void sendBoardApprovedEmails() {
    var approvedByBoardCapitalTransfers =
        capitalTransferContractRepository.findAllByState(APPROVED);

    if (approvedByBoardCapitalTransfers.isEmpty()) {
      log.info("No APPROVED capital transfer contracts found");
      return;
    }
    log.info(
        "Need to send {} emails about capital transfer contract board approval",
        approvedByBoardCapitalTransfers.size() * 2);

    for (var transfer : approvedByBoardCapitalTransfers) {
      try {
        log.info("Sending approval emails for capital transfer (id={})", transfer.getId());
        capitalTransferContractService.sendContractEmail(
            transfer.getBuyer().getUser(), CAPITAL_TRANSFER_APPROVED_BY_BOARD, transfer);
        capitalTransferContractService.sendContractEmail(
            transfer.getSeller().getUser(), CAPITAL_TRANSFER_APPROVED_BY_BOARD, transfer);

        capitalTransferContractService.updateStateBySystem(transfer.getId(), APPROVED_AND_NOTIFIED);
      } catch (Exception e) {
        log.error(
            "Failed to send email for capital transfer (id={}) with exception: {}",
            transfer.getId(),
            e.toString());
      }
    }
  }
}
