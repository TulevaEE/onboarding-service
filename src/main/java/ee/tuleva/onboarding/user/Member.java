package ee.tuleva.onboarding.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Past;
import java.io.Serializable;
import java.time.Instant;

@Data
@Builder
@Entity
@Table(name = "member")
@AllArgsConstructor
@NoArgsConstructor
public class Member implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private Integer memberNumber;

    @NotNull
    @Past
    private Instant createdDate;

}
