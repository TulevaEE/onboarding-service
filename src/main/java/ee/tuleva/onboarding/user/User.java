package ee.tuleva.onboarding.user;

import com.fasterxml.jackson.annotation.JsonView;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Email;
import org.hibernate.validator.constraints.NotBlank;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Past;
import javax.validation.constraints.Size;
import java.io.Serializable;
import java.time.Instant;

@Data
@Builder
@Entity
@Table(name = "users")
@AllArgsConstructor
@NoArgsConstructor
public class User implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@NotBlank
	@Size(min = 11, max = 11)
	private String personalCode;

	@NotNull
	@Email
	private String email;

	private String phoneNumber;

	@JsonView(UserView.Public.class)
	@NotBlank
	private String firstName;

	@JsonView(UserView.Public.class)
	@NotBlank
	private String lastName;

	@NotNull
	private Integer memberNumber;

	@NotNull
	@Past
	private Instant createdDate;

	@NotNull
	private Instant updatedDate;

	@NotNull
	private Boolean active;

	@JsonView(UserView.Public.class)
	public int getAge() {
		return PersonalCode.getAge(personalCode);
	}

}
