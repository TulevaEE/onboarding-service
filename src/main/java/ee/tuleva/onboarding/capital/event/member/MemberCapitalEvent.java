package ee.tuleva.onboarding.capital.event.member;

import ee.tuleva.onboarding.user.member.Member;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Data
@Builder
@Entity
@Table(name = "member_capital_event")
@AllArgsConstructor
@NoArgsConstructor
public class MemberCapitalEvent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @NotNull
    private MemberCapitalEventType type;

    @NotNull
    private BigDecimal fiatValue;

    @NotNull
    private BigDecimal ownershipUnitAmount;

}
