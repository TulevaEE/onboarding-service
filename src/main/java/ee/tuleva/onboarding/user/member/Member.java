package ee.tuleva.onboarding.user.member;

import com.fasterxml.jackson.annotation.JsonIgnore;
import ee.tuleva.onboarding.user.User;
import java.io.Serializable;
import java.time.Instant;
import javax.persistence.*;
import javax.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.Where;

@Data
@Builder
@Entity
@Table(name = "member")
@AllArgsConstructor
@NoArgsConstructor
@ToString(exclude = {"user"})
@Where(clause = "active = true")
public class Member implements Serializable {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @JsonIgnore
  @OneToOne(fetch = FetchType.LAZY, optional = false)
  @JoinColumn(name = "user_id", nullable = false)
  private User user;

  @NotNull private Integer memberNumber;

  @NotNull private Instant createdDate;

  @NotNull @Builder.Default private Boolean active = true;

  @PrePersist
  protected void onCreate() {
    createdDate = Instant.now();
  }
}
