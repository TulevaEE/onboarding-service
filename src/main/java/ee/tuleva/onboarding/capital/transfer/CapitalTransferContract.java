package ee.tuleva.onboarding.capital.transfer;

import static org.hibernate.type.SqlTypes.JSON;

import ee.tuleva.onboarding.capital.event.member.MemberCapitalEventType;
import ee.tuleva.onboarding.capital.transfer.iban.ValidIban;
import ee.tuleva.onboarding.time.ClockHolder;
import ee.tuleva.onboarding.user.User;
import ee.tuleva.onboarding.user.member.Member;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;

@Entity
@Table(name = "capital_transfer_contract")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CapitalTransferContract {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @NotNull
  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "seller_id")
  private Member seller;

  @NotNull
  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "buyer_id")
  private Member buyer;

  @NotNull @ValidIban private String iban;

  @JdbcTypeCode(JSON)
  @NotNull
  private List<CapitalTransferAmount> transferAmounts;

  @NotNull
  @Enumerated(EnumType.STRING)
  private CapitalTransferContractState state;

  @NotNull private byte[] originalContent;

  private byte[] digiDocContainer;

  private LocalDateTime createdAt;

  private LocalDateTime updatedAt;

  @PrePersist
  protected void onCreate() {
    LocalDateTime now = LocalDateTime.now(ClockHolder.getClock());
    createdAt = now;
    updatedAt = now;
  }

  @PreUpdate
  protected void onUpdate() {
    updatedAt = LocalDateTime.now(ClockHolder.getClock());
  }

  private void requireState(CapitalTransferContractState requiredStatus) {
    if (this.state != requiredStatus) {
      throw new IllegalStateException(
          "Action requires state " + requiredStatus + ", but current state is " + this.state + ".");
    }
  }

  public CapitalTransferContract assignBuyer(Member buyer) {
    this.setBuyer(buyer);
    return this;
  }

  public CapitalTransferContract signBySeller(byte[] container) {
    requireState(CapitalTransferContractState.CREATED);
    this.setDigiDocContainer(container);
    this.setState(CapitalTransferContractState.SELLER_SIGNED);
    return this;
  }

  public CapitalTransferContract signByBuyer(byte[] updatedContainer) {
    requireState(CapitalTransferContractState.SELLER_SIGNED);
    this.setDigiDocContainer(updatedContainer);
    this.setState(CapitalTransferContractState.BUYER_SIGNED);
    return this;
  }

  public CapitalTransferContract confirmPaymentByBuyer() {
    requireState(CapitalTransferContractState.BUYER_SIGNED);
    this.setState(CapitalTransferContractState.PAYMENT_CONFIRMED_BY_BUYER);
    return this;
  }

  public CapitalTransferContract confirmPaymentBySeller() {
    requireState(CapitalTransferContractState.PAYMENT_CONFIRMED_BY_BUYER);
    this.setState(CapitalTransferContractState.PAYMENT_CONFIRMED_BY_SELLER);
    return this;
  }

  public CapitalTransferContract approvedAndNotified() {
    requireState(CapitalTransferContractState.APPROVED);
    this.setState(CapitalTransferContractState.APPROVED_AND_NOTIFIED);
    return this;
  }

  public CapitalTransferContract cancel() {
    if (this.state == CapitalTransferContractState.APPROVED
        || this.state == CapitalTransferContractState.APPROVED_AND_NOTIFIED
        || this.state == CapitalTransferContractState.EXECUTED) {
      throw new IllegalStateException(
          "Cannot cancel a contract that is already approved or executed.");
    }
    this.setState(CapitalTransferContractState.CANCELLED);
    return this;
  }

  public boolean canBeAccessedBy(User user) {
    var isSeller = user.getMemberOrThrow().getId().equals(seller.getId());
    var isBuyer = user.getMemberOrThrow().getId().equals(buyer.getId());

    return isBuyer || isSeller;
  }

  public BigDecimal getTotalPrice() {
    return transferAmounts.stream()
        .map(CapitalTransferAmount::price)
        .reduce(BigDecimal.ZERO, BigDecimal::add);
  }

  public record CapitalTransferAmount(
      MemberCapitalEventType type, BigDecimal price, BigDecimal bookValue) {}
}
