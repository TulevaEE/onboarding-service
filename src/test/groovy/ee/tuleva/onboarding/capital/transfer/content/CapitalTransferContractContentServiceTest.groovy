package ee.tuleva.onboarding.capital.transfer.content

import au.com.origin.snapshots.Expect
import au.com.origin.snapshots.annotations.SnapshotName
import au.com.origin.snapshots.spock.EnableSnapshots
import ee.tuleva.onboarding.capital.transfer.CapitalTransferContract
import ee.tuleva.onboarding.user.User
import ee.tuleva.onboarding.user.member.Member
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import spock.lang.Specification

import java.time.LocalDateTime

import static ee.tuleva.onboarding.auth.UserFixture.sampleUser
import static ee.tuleva.onboarding.capital.event.member.MemberCapitalEventType.CAPITAL_PAYMENT
import static ee.tuleva.onboarding.capital.event.member.MemberCapitalEventType.MEMBERSHIP_BONUS
import static ee.tuleva.onboarding.capital.transfer.CapitalTransferContractState.CREATED
import static ee.tuleva.onboarding.user.MemberFixture.memberFixture

@SpringBootTest
@EnableSnapshots
class CapitalTransferContractContentServiceTest extends Specification {

  @Autowired
  CapitalTransferContractContentService contentService

  private Expect expect

  @SnapshotName("capital_transfer_contract_member_capital")
  def "generates HTML for member capital transfer contract"() {
    given:
    User sellerUser = sampleUser()
        .firstName("John")
        .lastName("Seller")
        .personalCode("37605030299")
        .email("seller@example.com")
        .build()
    User buyerUser = sampleUser()
        .id(2L)
        .personalCode("60001019906")
        .firstName("Mari")
        .lastName("Buyer")
        .email("buyer@example.com")
        .build()

    Member seller = memberFixture().user(sellerUser).memberNumber(1001).build()
    Member buyer = memberFixture().user(buyerUser).memberNumber(1002).build()

    CapitalTransferContract contract = CapitalTransferContract.builder()
        .seller(seller)
        .buyer(buyer)
        .iban("EE471000001020145685")
        .transferAmounts(
            List.of(
                new CapitalTransferContract.CapitalTransferAmount(
                    CAPITAL_PAYMENT, new BigDecimal("2500.0"), new BigDecimal("1250.0"))))
        .state(CREATED)
        .createdAt(LocalDateTime.of(2023, 12, 15, 10, 30))
        .build()

    when:
    String html = contentService.generateContractHtml(contract)

    then:
    expect.toMatchSnapshot(html)
  }

  @SnapshotName("capital_transfer_contract_member_bonus")
  def "generates HTML for member bonus transfer contract"() {
    given:
    User sellerUser = sampleUser()
        .firstName("Jane")
        .lastName("Seller")
        .personalCode("48001010001")
        .email("jane@example.com")
        .build()
    User buyerUser = sampleUser()
        .id(2L)
        .personalCode("39001010002")
        .firstName("Peeter")
        .lastName("Buyer")
        .email("peeter@example.com")
        .build()

    Member seller = memberFixture().user(sellerUser).memberNumber(2001).build()
    Member buyer = memberFixture().user(buyerUser).memberNumber(2002).build()

    CapitalTransferContract contract = CapitalTransferContract.builder()
        .seller(seller)
        .buyer(buyer)
        .iban("EE211722839095030454")
        .transferAmounts(
            List.of(
                new CapitalTransferContract.CapitalTransferAmount(
                    MEMBERSHIP_BONUS, new BigDecimal("200.0"), new BigDecimal("100.0"))))
        .state(CREATED)
        .createdAt(LocalDateTime.of(2024, 1, 20, 14, 45))
        .build()

    when:
    String html = contentService.generateContractHtml(contract)

    then:
    expect.toMatchSnapshot(html)
  }

  @SnapshotName("capital_transfer_contract_multiple_amounts")
  def "generates HTML for multiple capital transfer amounts"() {
    given:
    User sellerUser = sampleUser()
        .firstName("Alice")
        .lastName("MultiSeller")
        .personalCode("50001010003")
        .email("alice@example.com")
        .build()
    User buyerUser = sampleUser()
        .id(2L)
        .personalCode("40001010004")
        .firstName("Bob")
        .lastName("MultiBuyer")
        .email("bob@example.com")
        .build()

    Member seller = memberFixture().user(sellerUser).memberNumber(3001).build()
    Member buyer = memberFixture().user(buyerUser).memberNumber(3002).build()

    CapitalTransferContract contract = CapitalTransferContract.builder()
        .seller(seller)
        .buyer(buyer)
        .iban("EE211722839095030454")
        .transferAmounts(
            List.of(
                new CapitalTransferContract.CapitalTransferAmount(
                    CAPITAL_PAYMENT, new BigDecimal("2500"), new BigDecimal("1250.0")),
                new CapitalTransferContract.CapitalTransferAmount(
                    MEMBERSHIP_BONUS, new BigDecimal("400"), new BigDecimal("200.0"))))
        .state(CREATED)
        .createdAt(LocalDateTime.of(2024, 3, 10, 16, 15))
        .build()

    when:
    String html = contentService.generateContractHtml(contract)

    then:
    expect.toMatchSnapshot(html)
  }

  def "generates contract content as bytes"() {
    given:
    User sellerUser = sampleUser().build()
    User buyerUser = sampleUser()
        .id(2L)
        .personalCode("60001019906")
        .firstName("Mari")
        .lastName("Buyer")
        .email("buyer@example.com")
        .build()

    Member seller = memberFixture().user(sellerUser).memberNumber(1001).build()
    Member buyer = memberFixture().user(buyerUser).memberNumber(1002).build()

    CapitalTransferContract contract = CapitalTransferContract.builder()
        .seller(seller)
        .buyer(buyer)
        .iban("EE211722839095030454")
        .transferAmounts(
            List.of(
                new CapitalTransferContract.CapitalTransferAmount(
                    CAPITAL_PAYMENT, new BigDecimal("100.0"), new BigDecimal("1250.0"))))
        .state(CREATED)
        .createdAt(LocalDateTime.of(2023, 12, 15, 10, 30))
        .build()

    when:
    byte[] result = contentService.generateContractContent(contract)

    then:
    result.length > 0
    new String(result).contains("Liikmekapitali võõrandamise avaldus")
  }
}
