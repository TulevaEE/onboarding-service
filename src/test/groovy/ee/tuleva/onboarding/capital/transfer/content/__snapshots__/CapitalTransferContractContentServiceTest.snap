capital_transfer_contract_member_bonus=[
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Liikmekapitali müügileping</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            font-size: 12px;
            line-height: 1.4;
            margin: 0;
            padding: 40px;
            color: #333;
            background-color: #f5f5f5;
            display: flex;
            justify-content: center;
            min-height: 100vh;
        }
        .document {
            max-width: 21cm;
            width: 100%;
            background-color: white;
            box-shadow: 0 0 10px rgba(0,0,0,0.1);
            padding: 40px;
            box-sizing: border-box;
        }
        .header {
            text-align: center;
            margin-bottom: 30px;
        }
        .contract-title {
            font-size: 18px;
            font-weight: bold;
            margin-bottom: 20px;
        }
        .section {
            margin-bottom: 20px;
        }
        .section-title {
            font-weight: bold;
            margin-bottom: 10px;
        }
        .party-info {
            margin-bottom: 15px;
        }
        .party-title {
            font-weight: bold;
            text-decoration: underline;
        }
        .contract-details {
            margin: 20px 0;
        }
        .detail-row {
            display: flex;
            margin-bottom: 8px;
        }
        .detail-label {
            width: 180px;
            font-weight: bold;
        }
        .detail-value {
            flex: 1;
            text-align: right;
        }
        .transfer-amount-row {
            margin-left: 20px;
        }
        .transfer-amount-row .detail-label {
            width: 160px;
        }
        .signature-section {
            margin-top: 40px;
            display: flex;
            justify-content: space-between;
        }
        .signature-block {
            width: 200px;
            text-align: center;
        }
        .signature-line {
            border-bottom: 1px solid #333;
            margin-bottom: 5px;
            height: 30px;
        }
        .terms {
            margin: 20px 0;
            font-size: 11px;
        }
        .terms ol {
            padding-left: 20px;
        }
        .terms li {
            margin-bottom: 8px;
        }
    </style>
</head>
<body>
<div class="document">
    <div class="header">
        <div class="contract-title">LIIKMEKAPITALI MÜÜGILEPING</div>
        <div>Koostatud: <span>20.01.2024 14:45</span></div>
    </div>

    <div class="section">
        <div class="section-title">LEPINGU POOLED</div>
        
        <div class="party-info">
            <div class="party-title">MÜÜJA:</div>
            <div>Jane Seller</div>
            <div>Isikukood: <span>48001010001</span></div>
            <div>Liikmenumber: <span>2001</span></div>
        </div>

        <div class="party-info">
            <div class="party-title">OSTJA:</div>
            <div>Peeter Buyer</div>
            <div>Isikukood: <span>39001010002</span></div>
            <div>Liikmenumber: <span>2002</span></div>
        </div>
    </div>

    <div class="section">
        <div class="section-title">MÜÜGIESE ÜKSIKASJAD</div>
        
        <div class="contract-details">
            <div class="detail-row">
                <div class="detail-label">Müüdav liikmekapital:</div>
                <div class="detail-value"><span>100.00</span> EUR</div>
            </div>
            
            <div class="detail-row transfer-amount-row">
                <div class="detail-label">– <span>Liikmeboonus</span>:</div>
                <div class="detail-value"><span>100.00</span> EUR</div>
            </div>
            
            <div class="detail-row">
                <div class="detail-label">Hinnaga:</div>
                <div class="detail-value"><span>200.00</span> EUR</div>
            </div>
            
            <div class="detail-row">
                <div class="detail-label">Ostja pangakonto:</div>
                <div class="detail-value">EE382200221020145685</div>
            </div>
        </div>
    </div>

    <div class="section">
        <div class="section-title">LEPINGU TINGIMUSED</div>
        <div class="terms">
            <ol>
                <li>Käesoleva lepinguga müüb müüja ja ostab ostja liikmekapitali osad vastavalt käesoleva lepingu punktis "Müügiese üksikasjad" toodud tingimustele.</li>
                <li>Müüja kinnitab, et tal on õigus müüa käesolevas lepingus nimetatud liikmekapitali osad.</li>
                <li>Ostja kohustub tasuma müügihinna lepingu allkirjastamisest 5 tööpäeva jooksul müüja poolt märgitud pangakontole.</li>
                <li>Osade omandiõigus läheb üle ostjale pärast müügihinna täielikku laekumist müüja arvele.</li>
                <li>Mõlemad pooled kohustuvad täitma kõik seadusest tulenevad nõuded liikmekapitali osade müügi osas.</li>
                <li>Lepingu muudatused on kehtivad ainult kirjalikus vormis ja mõlema poole allkirjastamisel.</li>
                <li>Lepingust tulenevad vaidlused lahendatakse läbirääkimiste teel või kohtulikku korda.</li>
            </ol>
        </div>
    </div>

    <div class="signature-section">
        <div class="signature-block">
            <div class="signature-line"></div>
            <div><strong>MÜÜJA</strong></div>
            <div>Jane Seller</div>
        </div>
        
        <div class="signature-block">
            <div class="signature-line"></div>
            <div><strong>OSTJA</strong></div>
            <div>Peeter Buyer</div>
        </div>
    </div>
</div>
</body>
</html>
]


capital_transfer_contract_member_capital=[
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Liikmekapitali müügileping</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            font-size: 12px;
            line-height: 1.4;
            margin: 0;
            padding: 40px;
            color: #333;
            background-color: #f5f5f5;
            display: flex;
            justify-content: center;
            min-height: 100vh;
        }
        .document {
            max-width: 21cm;
            width: 100%;
            background-color: white;
            box-shadow: 0 0 10px rgba(0,0,0,0.1);
            padding: 40px;
            box-sizing: border-box;
        }
        .header {
            text-align: center;
            margin-bottom: 30px;
        }
        .contract-title {
            font-size: 18px;
            font-weight: bold;
            margin-bottom: 20px;
        }
        .section {
            margin-bottom: 20px;
        }
        .section-title {
            font-weight: bold;
            margin-bottom: 10px;
        }
        .party-info {
            margin-bottom: 15px;
        }
        .party-title {
            font-weight: bold;
            text-decoration: underline;
        }
        .contract-details {
            margin: 20px 0;
        }
        .detail-row {
            display: flex;
            margin-bottom: 8px;
        }
        .detail-label {
            width: 180px;
            font-weight: bold;
        }
        .detail-value {
            flex: 1;
            text-align: right;
        }
        .transfer-amount-row {
            margin-left: 20px;
        }
        .transfer-amount-row .detail-label {
            width: 160px;
        }
        .signature-section {
            margin-top: 40px;
            display: flex;
            justify-content: space-between;
        }
        .signature-block {
            width: 200px;
            text-align: center;
        }
        .signature-line {
            border-bottom: 1px solid #333;
            margin-bottom: 5px;
            height: 30px;
        }
        .terms {
            margin: 20px 0;
            font-size: 11px;
        }
        .terms ol {
            padding-left: 20px;
        }
        .terms li {
            margin-bottom: 8px;
        }
    </style>
</head>
<body>
<div class="document">
    <div class="header">
        <div class="contract-title">LIIKMEKAPITALI MÜÜGILEPING</div>
        <div>Koostatud: <span>15.12.2023 10:30</span></div>
    </div>

    <div class="section">
        <div class="section-title">LEPINGU POOLED</div>
        
        <div class="party-info">
            <div class="party-title">MÜÜJA:</div>
            <div>John Seller</div>
            <div>Isikukood: <span>37605030299</span></div>
            <div>Liikmenumber: <span>1001</span></div>
        </div>

        <div class="party-info">
            <div class="party-title">OSTJA:</div>
            <div>Mari Buyer</div>
            <div>Isikukood: <span>60001019906</span></div>
            <div>Liikmenumber: <span>1002</span></div>
        </div>
    </div>

    <div class="section">
        <div class="section-title">MÜÜGIESE ÜKSIKASJAD</div>
        
        <div class="contract-details">
            <div class="detail-row">
                <div class="detail-label">Müüdav liikmekapital:</div>
                <div class="detail-value"><span>1 250.00</span> EUR</div>
            </div>
            
            <div class="detail-row transfer-amount-row">
                <div class="detail-label">– <span>Rahaline panus</span>:</div>
                <div class="detail-value"><span>1 250.00</span> EUR</div>
            </div>
            
            <div class="detail-row">
                <div class="detail-label">Hinnaga:</div>
                <div class="detail-value"><span>2 500.00</span> EUR</div>
            </div>
            
            <div class="detail-row">
                <div class="detail-label">Ostja pangakonto:</div>
                <div class="detail-value">EE471000001020145685</div>
            </div>
        </div>
    </div>

    <div class="section">
        <div class="section-title">LEPINGU TINGIMUSED</div>
        <div class="terms">
            <ol>
                <li>Käesoleva lepinguga müüb müüja ja ostab ostja liikmekapitali osad vastavalt käesoleva lepingu punktis "Müügiese üksikasjad" toodud tingimustele.</li>
                <li>Müüja kinnitab, et tal on õigus müüa käesolevas lepingus nimetatud liikmekapitali osad.</li>
                <li>Ostja kohustub tasuma müügihinna lepingu allkirjastamisest 5 tööpäeva jooksul müüja poolt märgitud pangakontole.</li>
                <li>Osade omandiõigus läheb üle ostjale pärast müügihinna täielikku laekumist müüja arvele.</li>
                <li>Mõlemad pooled kohustuvad täitma kõik seadusest tulenevad nõuded liikmekapitali osade müügi osas.</li>
                <li>Lepingu muudatused on kehtivad ainult kirjalikus vormis ja mõlema poole allkirjastamisel.</li>
                <li>Lepingust tulenevad vaidlused lahendatakse läbirääkimiste teel või kohtulikku korda.</li>
            </ol>
        </div>
    </div>

    <div class="signature-section">
        <div class="signature-block">
            <div class="signature-line"></div>
            <div><strong>MÜÜJA</strong></div>
            <div>John Seller</div>
        </div>
        
        <div class="signature-block">
            <div class="signature-line"></div>
            <div><strong>OSTJA</strong></div>
            <div>Mari Buyer</div>
        </div>
    </div>
</div>
</body>
</html>
]


capital_transfer_contract_multiple_amounts=[
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Liikmekapitali müügileping</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            font-size: 12px;
            line-height: 1.4;
            margin: 0;
            padding: 40px;
            color: #333;
            background-color: #f5f5f5;
            display: flex;
            justify-content: center;
            min-height: 100vh;
        }
        .document {
            max-width: 21cm;
            width: 100%;
            background-color: white;
            box-shadow: 0 0 10px rgba(0,0,0,0.1);
            padding: 40px;
            box-sizing: border-box;
        }
        .header {
            text-align: center;
            margin-bottom: 30px;
        }
        .contract-title {
            font-size: 18px;
            font-weight: bold;
            margin-bottom: 20px;
        }
        .section {
            margin-bottom: 20px;
        }
        .section-title {
            font-weight: bold;
            margin-bottom: 10px;
        }
        .party-info {
            margin-bottom: 15px;
        }
        .party-title {
            font-weight: bold;
            text-decoration: underline;
        }
        .contract-details {
            margin: 20px 0;
        }
        .detail-row {
            display: flex;
            margin-bottom: 8px;
        }
        .detail-label {
            width: 180px;
            font-weight: bold;
        }
        .detail-value {
            flex: 1;
            text-align: right;
        }
        .transfer-amount-row {
            margin-left: 20px;
        }
        .transfer-amount-row .detail-label {
            width: 160px;
        }
        .signature-section {
            margin-top: 40px;
            display: flex;
            justify-content: space-between;
        }
        .signature-block {
            width: 200px;
            text-align: center;
        }
        .signature-line {
            border-bottom: 1px solid #333;
            margin-bottom: 5px;
            height: 30px;
        }
        .terms {
            margin: 20px 0;
            font-size: 11px;
        }
        .terms ol {
            padding-left: 20px;
        }
        .terms li {
            margin-bottom: 8px;
        }
    </style>
</head>
<body>
<div class="document">
    <div class="header">
        <div class="contract-title">LIIKMEKAPITALI MÜÜGILEPING</div>
        <div>Koostatud: <span>10.03.2024 16:15</span></div>
    </div>

    <div class="section">
        <div class="section-title">LEPINGU POOLED</div>
        
        <div class="party-info">
            <div class="party-title">MÜÜJA:</div>
            <div>Alice MultiSeller</div>
            <div>Isikukood: <span>50001010003</span></div>
            <div>Liikmenumber: <span>3001</span></div>
        </div>

        <div class="party-info">
            <div class="party-title">OSTJA:</div>
            <div>Bob MultiBuyer</div>
            <div>Isikukood: <span>40001010004</span></div>
            <div>Liikmenumber: <span>3002</span></div>
        </div>
    </div>

    <div class="section">
        <div class="section-title">MÜÜGIESE ÜKSIKASJAD</div>
        
        <div class="contract-details">
            <div class="detail-row">
                <div class="detail-label">Müüdav liikmekapital:</div>
                <div class="detail-value"><span>1 450.00</span> EUR</div>
            </div>
            
            <div class="detail-row transfer-amount-row">
                <div class="detail-label">– <span>Rahaline panus</span>:</div>
                <div class="detail-value"><span>1 250.00</span> EUR</div>
            </div>
            
            <div class="detail-row transfer-amount-row">
                <div class="detail-label">– <span>Liikmeboonus</span>:</div>
                <div class="detail-value"><span>200.00</span> EUR</div>
            </div>
            
            <div class="detail-row">
                <div class="detail-label">Hinnaga:</div>
                <div class="detail-value"><span>2 900.00</span> EUR</div>
            </div>
            
            <div class="detail-row">
                <div class="detail-label">Ostja pangakonto:</div>
                <div class="detail-value">EE123456789012345678</div>
            </div>
        </div>
    </div>

    <div class="section">
        <div class="section-title">LEPINGU TINGIMUSED</div>
        <div class="terms">
            <ol>
                <li>Käesoleva lepinguga müüb müüja ja ostab ostja liikmekapitali osad vastavalt käesoleva lepingu punktis "Müügiese üksikasjad" toodud tingimustele.</li>
                <li>Müüja kinnitab, et tal on õigus müüa käesolevas lepingus nimetatud liikmekapitali osad.</li>
                <li>Ostja kohustub tasuma müügihinna lepingu allkirjastamisest 5 tööpäeva jooksul müüja poolt märgitud pangakontole.</li>
                <li>Osade omandiõigus läheb üle ostjale pärast müügihinna täielikku laekumist müüja arvele.</li>
                <li>Mõlemad pooled kohustuvad täitma kõik seadusest tulenevad nõuded liikmekapitali osade müügi osas.</li>
                <li>Lepingu muudatused on kehtivad ainult kirjalikus vormis ja mõlema poole allkirjastamisel.</li>
                <li>Lepingust tulenevad vaidlused lahendatakse läbirääkimiste teel või kohtulikku korda.</li>
            </ol>
        </div>
    </div>

    <div class="signature-section">
        <div class="signature-block">
            <div class="signature-line"></div>
            <div><strong>MÜÜJA</strong></div>
            <div>Alice MultiSeller</div>
        </div>
        
        <div class="signature-block">
            <div class="signature-line"></div>
            <div><strong>OSTJA</strong></div>
            <div>Bob MultiBuyer</div>
        </div>
    </div>
</div>
</body>
</html>
]